### PR TITLE
Add Support team process

### PR DESCRIPTION
**Pull request by nickwood**

### CODE REVIEW

File: process/support_team.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+* If it is a larger piece of work, and/or non-urgent then it can be moved to the customer's backlog in Trello.
+
+Each Customer has their own Trello board for prioritising work. Issues are grouped differently for each customer but each board has the following 3 columns:
+1. 'New Issues' this is anything that's been backlogged from a Groove issue.
+2. 'Up next' issues will be automatically copied to our Support Epics board
+3. 'Recently Completed' issues are things which have been completed in the last week or so.
+
+During a weekly (bi-weekly for some customers) call with each customer Nick and the custmer will agree to a tentative commitment of work to be completed the following week by the support team. These issues will be moved to the 'Up Next' column, and copied to the Support Epics board.
+
+Every Monday everyone working on support that week will have a planning session to divide up the work for our customers across the team. Following this session Nick will follow up with each costomer to confirm what's going to be done that week.
```
Comment (chrisblackburn) [inline process/support_team.md]: I'm not at all averse to the Monday get together. That said, I wonder if planning out a week at a time discourages working on more meaty epics. Something we were musing, I think when you were on paternity, is whether there'd be something nice about doing fewer, longer spells of working for each customer. So trying to chunk 3, 5, 10 day epics completely focused on one customer for a pair?

**Comment by chrisblackburn** [comment]
As we move toward trying to chunk work in to more epic-like chunks, I still wonder whether showcases should start playing a bigger part in our support workflow. So having the team plan out these epics, set a showcase, ideally with the customer, or with an internal 'stakeholder' if every couple of days. We've seen success with the regular showcases helping to focus our efforts towards a delivery goal, and if nothing else provide some sanity checkpoint if things start overrunning our estimations as to how we should proceed

File: process/support_team.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+Every day support team members should attend the daily standup to report their progress ontheir support assignments, and flag up any issues or impediments which may impact the team's commitment that week.
+
+Once an issue is complete, the customer should be notified and the issue moved to the 'Done' column for the current week.
```
Comment (chrisblackburn) [inline process/support_team.md]: As a language thing, keen to canvas opinion on whether 'issue' is a loaded term. Particularly where we're trying to refocus efforts around proactive problem solving

File: process/support_team.md (hunk 1 above)
Comment (nickwood) [inline process/support_team.md]: Yep that is indeed the goal, though I haven't explicitly called it out here. Aiming for each pair to spend at least 3 days on something 'meaty' each week

File: process/support_team.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+If you're working on support in a week (or part of a week) then you should:
+ * Attend the planning session at the start of the week, please come armed with the number of days you can commit to support that week.
+ * Attend stand-ups on days you are doing support work.
+ * Where possible we like to pair on issues, especially for our more esoteric projects, in order to spread domain knowledge throughout the company, so come willing and able to do so.
+
+### The Triage Team
+
+As part of your support week you might want to volunteer to be on the triage team. This generally consists of one of Nick, Luke, Chris, Rory plus one other person.
+
+The Triage Team rotates every week, and it's their responsibility to ensure that anything new in Groove is resolved within 48 hours (noting that moving to backlog is a suitable resolution for non-urgent issues).
```
Comment (chrisblackburn) [inline process/support_team.md]: At least on Friday, Luke and I (as #teamTriage) were discussing whether the triage team should be clearing down this backlog once in the morning and once in the afternoon? I guess time will tell if this is workable when we have more customers using this channel, but it feels like a good customer experience to at least have a human acknowledge / run initial diagnostics within that sort of timeframe.

**Comment by nickwood** [comment]
Yep, the follow up on weekly call would be in addition to this show case. Showcases aren't currently part of the workflow, so I've not added them here

File: process/support_team.md (hunk 1 above)
Comment (craigjbass) [inline process/support_team.md]: My only worry with blocking into "Meaty" stuff is that WSJF specifically says target highest Urgency shortest job first. That could mean 4 disparate things that each take 3 hours each, which in turn delivers 80% of the business value (in terms of their backlog of tasks) to our client.

File: process/support_team.md (hunk 2 above)
Comment (nickwood) [inline process/support_team.md]: not too attached to it, could change to 'task' maybe? or or that feel too small?

File: process/support_team.md (hunk 3 above)
Comment (nickwood) [inline process/support_team.md]: Yeah would like first response to be same day, then resolution within 48. Will update

File: process/support_team.md (hunk 1 above)
Comment (nickwood) [inline process/support_team.md]: True, but in the new world these 3 hour jobs, could hopefully be tackled by the triage team, leaving the rest of support to pick up the meaty stuff

**Comment by RoryMacDonald** [comment]
Couple of bits:

* How should we handle triage / closing of support requests that require a customer reply? e.g Customer: This thing is broken. Made Tech: Can you ask Gioia to check the field names?.. then nada. I often find this is a point in which uncertainty creeps in and thing it would be sensible to have something in place to help customers know when we're waiting on them

* Trello added a new feature called custom fields the other day (see: https://blog.trello.com/trello-custom-fields/). Thought that could be interested for adding some structured meta data around tasks. One example might be when accepted this task (so we can track time from this to release time).

**Comment by craigjbass** [comment]
image

Terrible drawing but this is how I see a mature Kanban flow working with WiP limits and a defined commitment point.

**Comment by craigjbass** [comment]
> One example might be when accepted this task (so we can track time from this to release time).

For me Trello is not really a good enough tool for monitoring Kanban flow systems, you need CFD and Control charts as a minimum. I'm wondering if there is a plugin

**Comment by nickwood** [comment]
Well for small stuff this would all happen in Groove anyway, so Trello's
feature set is moot

--Nick

**Comment by lukemorton** [comment]
@nickwood Once you merge a version of this, I'll fold in https://gist.github.com/lukemorton/7d883451f3a4489fc86db36e5a65a09f

File: process/support_team.md [hunk 4]
```diff
@@ -4,43 +4,47 @@
...
- * Attend the planning session at the start of the week, please come armed with the number of days you can commit to support that week.
- * Attend stand-ups on days you are doing support work.
- * Where possible we like to pair on issues, especially for our more esoteric projects, in order to spread domain knowledge throughout the company, so come willing and able to do so.
+## Support Planning
 
-### The Triage Team
+During a weekly call (bi-weekly for some) with each customer Nick and the customer will agree to a commitment of work to be completed the following week by the support team. These issues will be moved to the 'Up Next' column, and copied to the Support Epics board.
 
-As part of your support week you might want to volunteer to be on the triage team. This generally consists of one of Nick, Luke, Chris, Rory plus one other person.
+By Thursday of each week the Support Epics board will have:
```
Comment (craigjbass) [inline process/support_team.md]: What is the definition of "reasonable". A top-down approach or a bottom-up approach?

File: process/support_team.md [hunk 5]
```diff
@@ -4,43 +4,47 @@
...
+ * A list of items that can be reasonably completed by support pairs in the upcoming week, and have a pair assigned to them.
+ * A list of 'Triage Team' items, which can reasonably be completed by memebers of the Triage Team around core Triage work.
 
-The Triage Team rotates every week, and it's their responsibility to ensure that anything new in Groove is resolved within 48 hours (noting that moving to backlog is a suitable resolution for non-urgent issues).
+## Working in a Support Pair
 
-Note that any new activity on Groove triggers a notification to the #support channel in Slack, so it's worth idling in there to save having to keep hitting refresh...
+If you're on a Support Pair then it is your responsibility to:
+ * Ensure you know at the start of each week what your pair will be working on, and have a goal in mind for the Scrum of Scrums.
+ * Book in at least one showcase with the customer that week so they can see what progress ahs been made towards the goal.
```
Comment (craigjbass) [inline process/support_team.md]: ahs -> has

File: process/support_team.md [hunk 6]
```diff
@@ -4,43 +4,47 @@
...
+ * At the end of each day record your time in Harvest against the relevant customer. You should also include a brief description of the work carried out.
+ * Communicate regulalarly with the customer especially around deployments so they can plan any testing they may want to do.
 
+## Working in Team Triage
 
+If you're working on Team Triage then it is your responsibility to:
+ * Know which days you are on Triage duty, and swap days in advance if you're unable to.
+ * Work towards the Triage goal of resolving everything within 48 hours (NB. Backlogging an issue is an accepted resolution)
+ * Call for help from others if there is more triage work than you can resonably handle
+ * Record all time spent in Harvest against the respoective customers
```
Comment (craigjbass) [inline process/support_team.md]: respoective -> respective

File: process/support_team.md [hunk 7]
```diff
@@ -4,43 +4,47 @@
...
+ * Book in at least one showcase with the customer that week so they can see what progress ahs been made towards the goal.
+ * At the end of each day record your time in Harvest against the relevant customer. You should also include a brief description of the work carried out.
+ * Communicate regulalarly with the customer especially around deployments so they can plan any testing they may want to do.
 
+## Working in Team Triage
 
+If you're working on Team Triage then it is your responsibility to:
+ * Know which days you are on Triage duty, and swap days in advance if you're unable to.
+ * Work towards the Triage goal of resolving everything within 48 hours (NB. Backlogging an issue is an accepted resolution)
+ * Call for help from others if there is more triage work than you can resonably handle
```
Comment (craigjbass) [inline process/support_team.md]: What is the internal mechanism which ensures this doesn't have a negative impact on other customer goals?

File: process/support_team.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+## Support Team Structure
+
+The support team doesn't have a fixed roster of staff and instead consists of a number of developers based availibility of staff and customer demand for support work. The support team is formed of:
+ * The **'Triage Team'**, a core rota of staff each contributing a day or more of their week to oversee intake of new issues, and fixing small and critical issues.
+ * A variable number of **'Support Pairs'**. These pairs are stable from week to week, and generally have an overarching goal which spans many weeks.
+
+## Support issue lifecycle
+
+All support issues begin life as an email into support@madetech.com. These are forwarded automatically into our GrooveHQ inbox to be sorted by the Triage Team. We aim to resolve all new issues within 48 hours with one of the following resolutions:
```
Comment (alex-min) [inline process/support_team.md]: .com or .co.uk? should we remove one of those?

File: process/support_team.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+ * A list of items that can be reasonably completed by support pairs in the upcoming week, and have a pair assigned to them.
+ * A list of 'Triage Team' items, which can reasonably be completed by memebers of the Triage Team around core Triage work.
+
+## Working in a Support Pair
+
+If you're on a Support Pair then it is your responsibility to:
+ * Ensure you know at the start of each week what your pair will be working on, and have a goal in mind for the Scrum of Scrums.
+ * Book in at least one showcase with the customer that week so they can see what progress has been made towards the goal.
+ * At the end of each day record your time in Harvest against the relevant customer. You should also include a brief description of the work carried out.
+ * Communicate regulalarly with the customer especially around deployments so they can plan any testing they may want to do.
```
Comment (alex-min) [inline process/support_team.md]: regulalarly -> regularly

File: process/support_team.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+Pending is for when waiting for a reply which is needed to progress. Closed is for when an issue is resolved/backlogged. Often a user will reply to the final message - this can just be closed unless it requires further action to be taken.
+
+### Who replies to customers, should it be same person or just whoever is on triage that day?
+
+In most cases whoever is on triage that day: outstanding triage work should be handed over from day to day. Occasionally specific domain knowledge may be required in which case it may be easier/more appropriate for that person to respond instead.
+
+### Who does larger quotations/estimation of work that is requested via groove?
+
+Ideally whoever is triage, but not everyone has the context of the project to be able to do it, so probably dependent on case. Ask if you're unsure
```
Comment (alex-min) [inline process/support_team.md]: missing final dot.

**Comment by alex-min** [comment]
:+1:, everything looks good!

File: process/support_team.md (hunk 8 above)
Comment (nickwood) [inline process/support_team.md]: both work, but .com is preferred (due to email routing causing duplicates if .co.uk is used)

**Pull request merged**